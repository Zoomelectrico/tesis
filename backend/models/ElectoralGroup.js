const mongoose = require("mongoose");

const electoralGroupSchema = new mongoose.Schema({
  denomination: {
    type: String,
    required: "Debe Proveer un nombre para el grupo electoral",
    trim: true,
    uppercase: true,
    unique: true
  },
  color: {
    type: String,
    required: "Debe proveer un Color",
    trim: true
  },
  colorHex: {
    type: String,
    required: "Debe proveer un Color",
    trim: true
  },
  number: {
    type: Number,
    required: "Debe proveer un Numero",
    trim: true,
    unique: true
  },
  logo: {
    type: String,
    required: "Debe proveer un logotipo"
  },
  representative: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User"
  },
  electionYear: {
    type: Number,
    default: new Date().getFullYear()
  }
});

module.exports = mongoose.model("ElectoralGroup", electoralGroupSchema);
