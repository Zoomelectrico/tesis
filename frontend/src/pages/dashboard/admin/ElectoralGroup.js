import React, { useState, useEffect } from "react";
import {
  Container,
  Row,
  Col,
  Card,
  CardHeader,
  CardBody,
  Button
} from "reactstrap";

import { get } from "../../../utils";
import { Header, Toast, notify } from "../../../components";

const ElectoralGroup = ({ location }) => {
  const id = new URLSearchParams(location.query).get("id");
  const [demand, setDemand] = useState({ loading: true, demand: {} });

  useEffect(() => {
    const fetch = async () => {
      try {
        const data = await get(`demand/${id}`);
        setDemand({ loading: true, demand: data.demand });
      } catch (err) {
        notify(
          "Hubo un problema al cargar los datos. Refresque la Pagina",
          false
        );
      }
    };
    fetch();
  }, []);

  return (
    <>
      <Header />
      <Container className="mt--7">
        <Row>
          <Col sm="12">
            {demand.loading ? (
              <h2>Loading...</h2>
            ) : (
              <Card>
                <CardHeader>
                  <h2>{demand.electoralGroup.denomination}</h2>
                </CardHeader>
                <CardBody>{/** */}</CardBody>
              </Card>
            )}
          </Col>
        </Row>
      </Container>
      <Toast />
    </>
  );
};

export default ElectoralGroup;
