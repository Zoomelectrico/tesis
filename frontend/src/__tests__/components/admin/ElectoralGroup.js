import React from 'react';
import { shallow } from 'enzyme';
import ElectoralGroup from '../../../pages/dashboard/admin/ElectoralGroup';

describe('Electoral Group', () => {
  it('Render Correctly', () => {
    const wrapper = shallow(<ElectoralGroup />);
    expect(wrapper).toMatchSnapshot();
  });
});
