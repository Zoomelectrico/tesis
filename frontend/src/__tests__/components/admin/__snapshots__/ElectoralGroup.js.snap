// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Electoral Group Render Correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ElectoralGroup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <Container
          className="mt--7"
          tag="div"
        >
          <Row
            tag="div"
          >
            <Col
              sm="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Card
                tag="div"
              >
                <div
                  className="d-flex justify-content-center mb-3 p-4"
                >
                  <img
                    alt="uvote logo"
                    height="125px"
                    src={
                      Object {
                        "ReactComponent": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                        "__esModule": true,
                        "default": "logo-color.svg",
                      }
                    }
                    width="auto"
                  />
                </div>
                <h2
                  className="text-center"
                >
                  Loading ...
                </h2>
              </Card>
            </Col>
          </Row>
        </Container>,
        <Toast />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Row
            tag="div"
          >
            <Col
              sm="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Card
                tag="div"
              >
                <div
                  className="d-flex justify-content-center mb-3 p-4"
                >
                  <img
                    alt="uvote logo"
                    height="125px"
                    src={
                      Object {
                        "ReactComponent": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                        "__esModule": true,
                        "default": "logo-color.svg",
                      }
                    }
                    width="auto"
                  />
                </div>
                <h2
                  className="text-center"
                >
                  Loading ...
                </h2>
              </Card>
            </Col>
          </Row>,
          "className": "mt--7",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Col
              sm="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Card
                tag="div"
              >
                <div
                  className="d-flex justify-content-center mb-3 p-4"
                >
                  <img
                    alt="uvote logo"
                    height="125px"
                    src={
                      Object {
                        "ReactComponent": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                        "__esModule": true,
                        "default": "logo-color.svg",
                      }
                    }
                    width="auto"
                  />
                </div>
                <h2
                  className="text-center"
                >
                  Loading ...
                </h2>
              </Card>
            </Col>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Card
                tag="div"
              >
                <div
                  className="d-flex justify-content-center mb-3 p-4"
                >
                  <img
                    alt="uvote logo"
                    height="125px"
                    src={
                      Object {
                        "ReactComponent": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                        "__esModule": true,
                        "default": "logo-color.svg",
                      }
                    }
                    width="auto"
                  />
                </div>
                <h2
                  className="text-center"
                >
                  Loading ...
                </h2>
              </Card>,
              "sm": "12",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <div
                    className="d-flex justify-content-center mb-3 p-4"
                  >
                    <img
                      alt="uvote logo"
                      height="125px"
                      src={
                        Object {
                          "ReactComponent": Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "render": [Function],
                          },
                          "__esModule": true,
                          "default": "logo-color.svg",
                        }
                      }
                      width="auto"
                    />
                  </div>,
                  <h2
                    className="text-center"
                  >
                    Loading ...
                  </h2>,
                ],
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="uvote logo"
                      height="125px"
                      src={
                        Object {
                          "ReactComponent": Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "render": [Function],
                          },
                          "__esModule": true,
                          "default": "logo-color.svg",
                        }
                      }
                      width="auto"
                    />,
                    "className": "d-flex justify-content-center mb-3 p-4",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "uvote logo",
                      "height": "125px",
                      "src": Object {
                        "ReactComponent": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                        "__esModule": true,
                        "default": "logo-color.svg",
                      },
                      "width": "auto",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Loading ...",
                    "className": "text-center",
                  },
                  "ref": null,
                  "rendered": "Loading ...",
                  "type": "h2",
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <Container
            className="mt--7"
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Card
                  tag="div"
                >
                  <div
                    className="d-flex justify-content-center mb-3 p-4"
                  >
                    <img
                      alt="uvote logo"
                      height="125px"
                      src={
                        Object {
                          "ReactComponent": Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "render": [Function],
                          },
                          "__esModule": true,
                          "default": "logo-color.svg",
                        }
                      }
                      width="auto"
                    />
                  </div>
                  <h2
                    className="text-center"
                  >
                    Loading ...
                  </h2>
                </Card>
              </Col>
            </Row>
          </Container>,
          <Toast />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Row
              tag="div"
            >
              <Col
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Card
                  tag="div"
                >
                  <div
                    className="d-flex justify-content-center mb-3 p-4"
                  >
                    <img
                      alt="uvote logo"
                      height="125px"
                      src={
                        Object {
                          "ReactComponent": Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "render": [Function],
                          },
                          "__esModule": true,
                          "default": "logo-color.svg",
                        }
                      }
                      width="auto"
                    />
                  </div>
                  <h2
                    className="text-center"
                  >
                    Loading ...
                  </h2>
                </Card>
              </Col>
            </Row>,
            "className": "mt--7",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Col
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Card
                  tag="div"
                >
                  <div
                    className="d-flex justify-content-center mb-3 p-4"
                  >
                    <img
                      alt="uvote logo"
                      height="125px"
                      src={
                        Object {
                          "ReactComponent": Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "render": [Function],
                          },
                          "__esModule": true,
                          "default": "logo-color.svg",
                        }
                      }
                      width="auto"
                    />
                  </div>
                  <h2
                    className="text-center"
                  >
                    Loading ...
                  </h2>
                </Card>
              </Col>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Card
                  tag="div"
                >
                  <div
                    className="d-flex justify-content-center mb-3 p-4"
                  >
                    <img
                      alt="uvote logo"
                      height="125px"
                      src={
                        Object {
                          "ReactComponent": Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "render": [Function],
                          },
                          "__esModule": true,
                          "default": "logo-color.svg",
                        }
                      }
                      width="auto"
                    />
                  </div>
                  <h2
                    className="text-center"
                  >
                    Loading ...
                  </h2>
                </Card>,
                "sm": "12",
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <div
                      className="d-flex justify-content-center mb-3 p-4"
                    >
                      <img
                        alt="uvote logo"
                        height="125px"
                        src={
                          Object {
                            "ReactComponent": Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "render": [Function],
                            },
                            "__esModule": true,
                            "default": "logo-color.svg",
                          }
                        }
                        width="auto"
                      />
                    </div>,
                    <h2
                      className="text-center"
                    >
                      Loading ...
                    </h2>,
                  ],
                  "tag": "div",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <img
                        alt="uvote logo"
                        height="125px"
                        src={
                          Object {
                            "ReactComponent": Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "render": [Function],
                            },
                            "__esModule": true,
                            "default": "logo-color.svg",
                          }
                        }
                        width="auto"
                      />,
                      "className": "d-flex justify-content-center mb-3 p-4",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "uvote logo",
                        "height": "125px",
                        "src": Object {
                          "ReactComponent": Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "render": [Function],
                          },
                          "__esModule": true,
                          "default": "logo-color.svg",
                        },
                        "width": "auto",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Loading ...",
                      "className": "text-center",
                    },
                    "ref": null,
                    "rendered": "Loading ...",
                    "type": "h2",
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
